src/components/Auth/ResetPassword.tsx

```diff
@@ -0,0 +1,143 @@
+import React from 'react';
+import { useState, useCallback, useEffect } from 'react';
+import { useHistory } from 'react-router-dom';
+import { Form, Typography, Input, Button, notification } from 'antd';
+import styled from 'styled-components';
+import qs from 'query-string';
+
+import { useAuth } from '../../hooks';
+import { validatePassword } from './Auth';
+
+function ResetPassword() {
+  const [form] = Form.useForm();
+  const { authenticating, onPasswordReset, isPasswordResetted } = useAuth();
+  const [password, setPassword] = useState('');
+  const [confirmPassword, setConfirmPassword] = useState(' ');
+
+  const history = useHistory();
+
+  const q = qs.parse(document.location.search);
+  const token = q.token as string;
+  const login = (q.email as string) || '';
+
+  const handlePasswordChange = useCallback((e: any) => {
+    setPassword(e.target.value);
+  }, []);
+  const handleconfirmPasswordChange = useCallback((e: any) => {
+    setConfirmPassword(e.target.value);
+  }, []);
+
+  const resetSuccessfulNotification = () => {
+    notification.success({
+      message: 'Password has been successfully changed.',
+      description: 'Your password has been successfully changed. You can log in with your new password now.',
+      duration: 3.5,
+    });
+  };
+
+  const handlePasswordReset = useCallback(async () => {
+    await form.validateFields();
+    if (password === confirmPassword) {
+      onPasswordReset({ login, password, token });
+    }
+  }, [login, token, password, confirmPassword, onPasswordReset, form]);
+
+  const checkPasswordMatch = useCallback(() => {
+    if (!(password === confirmPassword)) {
+      return Promise.reject(new Error('Passwords must match'));
+    }
+  }, [password, confirmPassword]);
+
+  useEffect(() => {
+    if (isPasswordResetted) {
+      resetSuccessfulNotification();
+      history.push('/login/');
+    }
+  }, [history, isPasswordResetted]);
+
+  return (
+    <Container>
+      <Form layout="vertical">
+        <Typography.Title level={3} className="title">
+          {`Reset Password for ${login.replace(/\s/g, '')}`}
+        </Typography.Title>
+        <Form.Item
+          name="password"
+          rules={[
+            { required: true, message: 'Password is required' },
+            { validator: (_, value) => validatePassword(value) },
+          ]}
+        >
+          <Input
+            type="password"
+            size="large"
+            style={{ width: '240px' }}
+            placeholder="New Password"
+            disabled={authenticating}
```

Comment:
Why use `authenticating` here? Have you checked when it's `true`? Certainly not when you reset your pw. Just make a new state variable `resetting` and change it inside `handlePasswordReset`

Reply:
i decided to remove this loading logic and simplify the code. Currently after clicking reset password button, i validate and then redirect the user to log in page where either a) password changes + success not. shows or b) Password reset is unsuccessful, please request a new email not. displays. Because the loading takes less than a second and i didn't see the button changes into loading stage, i decided to remove this logic as well.

Reply:
but what does `authenticating` have to do with anything? you did not see the state change because `authenticating` is not `true` at any stage here. like I said, just make a new state variable and use that for `loading` and `disabled` states. see https://github.com/chronologic/automate-ui/pull/66/files#diff-3552ca91a19e0c106caff749da10e652f8281c1574611681e7672ca1fe14b23eR40